ui/page/transaction/transactions_page.go

```diff
@@ -147,163 +146,272 @@ func (pg *TransactionsPage) pageTitle(gtx C) D {
 	return txt.Layout(gtx)
 }
 
+func (pg *TransactionsPage) getAssetType() utils.AssetType {
+	if pg.selectedWallet == nil {
+		return utils.DCRWalletAsset
+	}
+	return pg.selectedWallet.GetAssetType()
+}
+
 func (pg *TransactionsPage) refreshAvailableTxType() {
-	wal := pg.WL.SelectedWallet.Wallet
 	items := []cryptomaterial.DropDownItem{}
-	_, keysinfo := components.TxPageDropDownFields(wal.GetAssetType(), pg.selectedTabIndex)
+	_, keysinfo := components.TxPageDropDownFields(pg.getAssetType(), pg.selectedTxCategoryTab)
 	for _, name := range keysinfo {
-		item := cryptomaterial.DropDownItem{}
-		item.Text = name
-		items = append(items, item)
+		items = append(items, cryptomaterial.DropDownItem{Text: name})
 	}
-
 	pg.txTypeDropDown = pg.Theme.DropDown(items, values.TxDropdownGroup, 2)
 
-	go func() {
-		countfn := func(fType int32) int {
-			count, _ := wal.CountTransactions(fType)
-			return count
+	// only show tx count for regular txs, not staking
+	if pg.selectedTxCategoryTab == 0 {
```

Comment:
A few lines before this, I saw `pg.txCategoryTab.SelectedSegment() != values.String(values.StrTxOverview)`. That's easier to understand. Please remove the `pg.selectedTxCategoryTab` variable.
```suggestion
	if pg.txCategoryTab.SelectedSegment() == values.String(values.StrTxOverview) {
```